broker/endpoints/Patient.js

```diff
@@ -174,40 +196,55 @@ async function createPatient(patient, head) {
 
   // Create RelatedPerson resource
   let resource = RelatedPerson.toFHIR(related);
-  let relatedID = (await axios.post(`/RelatedPerson`, resource)).data.id;
+  const relatedID = (await axios.post(`/RelatedPerson`, resource)).data.id;
 
-  // Create Patient resource with ID from PatientIdService
-  // and link to RelatedPerson
-  let patientID = `${prepend}${id}`;
   patient.link = `RelatedPerson/${relatedID}`;
+  
   resource = Patient.toFHIR(patient);
-  resource.id = patientID;
-  await axios.put(`/Patient/${patientID}`, resource);
+  
+  let patientID = {
+	  qr_code: uuid.v4()
+  }
+
+  // Add QR Code UUID to Patient Identifier list
+  if (!resource.hasOwnProperty("identifier")) resource.identifier = [];
+  resource.identifier.push({
+	  value: patientID.qr_code,
+	  assigner: {
+		  display: "massvaxx"
+	  },
+	  period: {
+		  start: new Date().toISOString()
+	  }
+  });
+  
+  const createdPatient = await axios.post(`/Patient`, resource);
+  patientID.resourceId = createdPatient.data.id;
 
   // If RelatedPerson was not created with a link to a patient,
   // update the resource to link to the new Patient
   if (head == null) {
     head = patientID;
     resource = RelatedPerson.toFHIR({
-      patient: `Patient/${head}`,
-      relationship: patient.relationship,
+      patient: `Patient/${head.resourceId}`,
```

Comment:
head should be the resource id for the head of the household. It's used earlier above in this method and would cause issues if it's an object.

Reply:
This did cause an issue, but it should be fixed now.